examples/boost/CMakeLists.txt

```diff
@@ -1,21 +1,39 @@
-cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.14...3.23)
 
 project(CPMExampleBoost)
 
 # ---- Create binary ----
 
 add_executable(CPMExampleBoost main.cpp)
-target_compile_features(CPMExampleBoost PRIVATE cxx_std_17)
+target_compile_features(CPMExampleBoost PRIVATE cxx_std_11)
+
+add_executable(CPMExampleAsio deferred.cpp)
+target_compile_features(CPMExampleAsio PRIVATE cxx_std_17)
 
 # ---- Dependencies ----
 
 include(../../cmake/CPM.cmake)
 
-CPMAddPackage(
-  NAME Boost
-  VERSION 1.77.0
-  GITHUB_REPOSITORY "boostorg/boost"
-  GIT_TAG "boost-1.77.0"
-)
+if(EXISTS "$ENV{CPM_SOURCE_CACHE}/boost_1_79_0")
+  set(FETCHCONTENT_SOURCE_DIR_BOOST
+      "$ENV{CPM_SOURCE_CACHE}/boost_1_79_0"
+      CACHE PATH "Boost source DIR"
+  )
+  set(CPM_Boost_SOURCE
+      ${FETCHCONTENT_SOURCE_DIR_BOOST}
+      CACHE PATH "Manual override"
+  )
+
+  # Workaround to compile only used Boost libs with orig. CMmakeLists.txt
+  add_subdirectory(${CPM_Boost_SOURCE} Boost EXCLUDE_FROM_ALL)
+else()
```

Comment:
Could you remove the `if` branch and leave only the CPM example? We want the examples to be as simple as possible.

Reply:
yes